controllers/kuadrant_controller_test.go

```diff
@@ -0,0 +1,101 @@
+//go:build integration
+
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	"github.com/kuadrant/kuadrant-operator/pkg/common"
+	"github.com/kuadrant/limitador-operator/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Kuadrant controller", func() {
+	var (
+		testNamespace string
+	)
+	const (
+		afterEachTimeOut = NodeTimeout(3 * time.Minute)
+		kuadrant         = "kuadrant-sample"
+	)
+	Context("Reconcile limitador resources", func() {
+		BeforeEach(func(ctx SpecContext) {
+			testNamespace = CreateNamespaceWithContext(ctx)
+			ApplyKuadrantCR(testNamespace)
+		})
+
+		AfterEach(func(ctx SpecContext) {
+			DeleteNamespaceCallbackWithContext(ctx, testNamespace)
+		}, afterEachTimeOut)
+		It("Copy configuration from Kuadrant CR to Limitador CR", func(ctx SpecContext) {
+			kObj := &kuadrantv1beta1.Kuadrant{}
+			lObj := &v1alpha1.Limitador{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: kuadrant, Namespace: testNamespace}, kObj)
```

Comment:
```suggestion
				err := k8sClient.Get(ctx, client.ObjectKey{Name: kuadrant, Namespace: testNamespace}, kObj)
```

nit: perhaps use ctx from `SpecContext` thoughout instead of creating new context ? Also do we want define a `SpecTimeout` for the `SpecContext` so that `Eventually` using this context will timeout after a certain point if it gets stuck ?

Reply:
It's not a nit. I missed that one when updating the PR to use `Spec'Context`. Good catch.

Reply:
Cool, any thoughts on the `SpecTimeout`'s ?

Reply:
I like the idea of the `SpecTimeout`. It is good to have. The only problem that I see is we don't record the length of each test running. So our timeouts are a best guess value. In the future if we start hitting the `SpecTimeout` do we just up the value or track down the reason for then hitting it. This is a cause of missing metrics about our tests.